task_schedules/task_schedule_promote_supplement.cfg

```diff
@@ -37,10 +37,8 @@ notify       javier.gonzalez@cfa.harvard.edu
 # Note the syntax 'exec <number> : cmd', which means that the given command is
 # executed only once for each <number> of times the task is executed.
 
-<task agasc_supplement_update>
+<task supplement_promotion>
       cron       * * * * *
       check_cron * * * * *
-      exec 1: agasc-supplement-update --log-level debug --output-dir /proj/sot/ska/data/agasc/rc --report
-      exec 1: agasc-mag-estimate-report --output-dir /proj/sot/ska/data/agasc/rc
-      exec 1: cp -fr /proj/sot/ska/data/agasc/rc/supplement_reports/weekly/latest/stars/* /proj/sot/ska/data/agasc/rc/supplement_reports/stars
+      exec 1: mv $ENV{SKA}/data/agasc/rc/promote/* $ENV{SKA}/data/agasc/
```

Comment:
I do not understand the question about the section. Do you mean that it could be merged with the staging task?

yes, I agree it is better to explicitly copy each file.

Reply:
Hah. Github looks to have edited the markup out of my comment.  I was trying to say a `<check>` section.

Reply:
I just set `<check>` off in the original comment with backticks.

Reply:
ah! I understand now. I think it is not needed, because if it fails, the return code will be non zero, and it sends the alert. That is if I change it to copy each file explicitly.

Reply:
sorry... I didn't think fast enough in my first response.

This setup is Tom's proposal, where this particular task schedule runs every week. If we specify the files explicitly, and If there is nothing to promote, the directory will be empty but this will still run and fail.

Reply:
The promotion directory could be empty without this being an error, so the commands should still succeed in this case. I think this can be done with two commands:
```
rsync -av $ENV{SKA}/data/agasc/rc/promote/ $ENV{SKA}/data/agasc/
rm -f $ENV{SKA}/data/agasc/rc/promote/*  # No error even if there are no files found
```

However, I think I would turn things around here:
- Change the `exec` line to run `agasc-supplement-tasks promote` command.
- Change the `promote` command to do the file moving, where it logs the files that got promoted and emits an informational message if there were no files to promote. This is in Python and it's easy.
- Have a regular weekly cron job (e.g. Sunday 1pm) that runs `task_schedule3.pl -config agasc/task_schedule_promote_supplement.cfg` to execute a staged promotion.

Going one step further to simplify, I would just not bother with having this separate `task_schedule_promote_supplement.cfg` task and instead have the `agasc-supplement-tasks promote` command be the first thing that the `task_schedule_update_supplement_rc` does. (and rename that to be `task_schedule_update_supplement` to be more generic).

However, if you prefer to have a separate task I'm OK with that.